### ISSUE TITLE
Istio Subset- Level traffic splitting cannot be used for Rolling Update

### ISSUE THREAD

**Issue by gyz2009** [argoproj/argo-rollouts#1405]
## Summary 
Update process does not Scale down the old RS, Its renewal process is more like blue-green.
It waits for all new ReplicaSet pods  to started and scale down the old ReplicaSet Pods to 0.
```yaml
strategy:
  canary:
    # maxSurge: "25%"
    # maxUnavailable: "25%"
    scaleDownDelaySeconds: 30
    trafficRouting:
      istio:
        virtualService:
          name: helloworld-vsvc # required
          routes:
            - primary # optional if there is a single route in VirtualService, required otherwise
        destinationRule:
          name: helloworld-destrule # required
          canarySubsetName: canary # required
          stableSubsetName: stable # required
    steps:
      - setWeight: 20
      - pause: {duration: 20s}
      - setWeight: 40
      - pause: {duration: 20s}
      - setWeight: 60
      - pause: {duration: 20s}
      - setWeight: 80
      - pause: {duration: 20s}
```

## Diagnostics

What version of Argo Rollouts are you running?
1.0.4

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by huikang** [comment]
Hi, @gyz2009 , could you paste the rollout status in the middle of an update, e.g., after step 5 (`setWeight: 60`)?

**Comment by huikang** [comment]
Never mind; I can reproduce this bug. Will look into this. Thanks.

**Comment by huikang** [comment]
@jessesuen , checking the following

https://github.com/argoproj/argo-rollouts/blob/cf19b5b88089adcf4cc0b493de2db232cfb9181c/utils/replicaset/canary.go#L110-L112

which seems cause the oldRs or ohterRS not to be scaled down. Is this the desired behavior for traffic routing?

**Comment by harikrongali** [comment]
@huikang  https://github.com/argoproj/argo-rollouts/pull/341/files#r363676969

**Comment by huikang** [comment]
@harikrongali , thanks so much for pointing me to the change. So this is the expected behavior.

@gyz2009 , I think we can close this one per design.

**Comment by gyz2009** [comment]
@huikang Ok, I have some RS Replicas 100+, due to insufficient resources, update will be in pending state.

**Comment by harikrongali** [comment]
@gyz2009 https://github.com/argoproj/argo-rollouts/issues/1029 solve what you are seeing and should be available in 1.1 release

**Comment by jessesuen** [comment]
Duplicate of https://github.com/argoproj/argo-rollouts/issues/1029

**Comment by gyz2009** [comment]
@harikrongali  I tested 1.1.0-RC1 and there was no change 
```yaml
strategy:
  canary: 
    dynamicStableScale: true
    # maxSurge: "25%"
    # maxUnavailable: "25%"
    # scaleDownDelaySeconds: 30
    trafficRouting:
      istio:
        virtualService:
          name: helloworld-vsvc
          routes:
            - primary
        destinationRule:
          name: helloworld 
          canarySubsetName: canary # required
          stableSubsetName: stable # required
    steps:
      - setCanaryScale:
          matchTrafficWeight: true
      - setCanaryScale:
          weight: 30
      - pause: {duration: 60s}
      - setCanaryScale:
          weight: 60
      - setCanaryScale:
          weight: 100

```
